packages/infra/lib/ihe-stack.ts

```diff
@@ -0,0 +1,92 @@
+import { CfnOutput, StackProps } from "aws-cdk-lib";
+import * as apig from "aws-cdk-lib/aws-apigateway";
+import * as cert from "aws-cdk-lib/aws-certificatemanager";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as r53 from "aws-cdk-lib/aws-route53";
+import * as r53_targets from "aws-cdk-lib/aws-route53-targets";
+import { Construct } from "constructs";
+import { EnvConfig } from "../config/env-config";
+import { createLambda } from "./shared/lambda";
+
+interface IHEStackProps extends StackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction: SnsAction | undefined;
+  lambdaLayers: lambda.ILayerVersion[];
+  certificate: cert.DnsValidatedCertificate;
+  publicZone: r53.IHostedZone;
+}
+
+export function createIHEStack(stack: Construct, props: IHEStackProps) {
+  //-------------------------------------------
+  // API Gateway
+  //-------------------------------------------
+  if (!props.config.iheSubdomain) {
+    throw new Error("Must define iheSubdomain if building the IHE stack!");
+  }
+
+  // Create the API Gateway
+  const api = new apig.RestApi(stack, "IHEAPIGateway", {
+    description: "Metriport IHE Gateway",
+    defaultCorsPreflightOptions: {
+      allowOrigins: ["*"],
+      allowHeaders: ["*"],
+    },
+  });
+
+  // add domain cert + record
+  const iheApiUrl = `${props.config.iheSubdomain}.${props.config.domain}`;
+  api.addDomainName("IHEAPIDomain", {
+    domainName: iheApiUrl,
+    certificate: props.certificate,
+    securityPolicy: apig.SecurityPolicy.TLS_1_2,
+  });
+  new r53.ARecord(stack, "IHEAPIDomainRecord", {
+    recordName: iheApiUrl,
+    zone: props.publicZone,
+    target: r53.RecordTarget.fromAlias(new r53_targets.ApiGateway(api)),
+  });
+
+  const iheLambda = createLambda({
+    stack: stack,
+    name: "IHE",
+    runtime: lambda.Runtime.NODEJS_18_X,
```

Comment:
If NABD, could leave this empty and use the default one (now, 18) to simplify migration down the road.

Reply:
will address